core-web/apps/dotcms-ui/src/app/view/components/dot-content-compare/store/dot-content-compare.store.ts

```diff
@@ -87,7 +87,29 @@ export class DotContentCompareStore extends ComponentStore<DotContentCompareStat
                                         return { contentType, contents };
                                     })
                                 );
-                        })
+                        }),
+                        // If the requested item is not found in the list of historical content versions, attempt to add it.
+                        switchMap(
+                            (value: {
+                                contentType: DotCMSContentType;
+                                contents: DotCMSContentlet[];
+                            }) => {
+                                return !value.contents.some(
+                                    (content) => content.inode === data.inode
+                                )
+                                    ? this.dotContentletService
+                                          .getContentletByInode(data.inode)
```

Comment:
How do we handle the `error` case If the request fails?

Note: We can use the: [DotHttpErrorManagerService](https://github.com/dotCMS/core/blob/master/core-web/libs/data-access/src/lib/dot-http-error-manager/dot-http-error-manager.service.ts)

Reply:
Thanks for the feedback, I try to improve it here https://github.com/dotCMS/core/pull/27663/commits/2349909b1927804cd165c7d53821e2ffafe66e10